### PR TITLE
Regression: Direct message creation by REST

### PR DESCRIPTION
**Pull request by ggazzo**
Closes https://github.com/RocketChat/Rocket.Chat/issues/17106

### CODE REVIEW

File: app/api/server/v1/im.js [hunk 1]
```diff
@@ -32,10 +32,18 @@ function findDirectMessageRoom(params, user) {
...
-		const findResult = findDirectMessageRoom(this.requestParams(), this.user);
+		const { username, usernames } = this.requestParams();
+
+		const users = username ? [username] : usernames && usernames.split(',').map((username) => username.trim());
+
+		if (!users) {
+			throw new Meteor.Error('error-room-not-found', 'The required "roomId", "username" or "usernames" param provided does not match any direct message');
+		}
+
+		const room = Meteor.runAsUser(this.userId, () => Meteor.call('createDirectMessage', ...users));
```
Comment (rodrigok) [inline app/api/server/v1/im.js:43]:
```suggestion
		const room = Meteor.call('createDirectMessage', ...users);
```
Run as user isn't needed anymore

File: app/api/server/v1/im.js [hunk 2]
```diff
@@ -32,10 +32,18 @@ function findDirectMessageRoom(params, user) {
 
 API.v1.addRoute(['dm.create', 'im.create'], { authRequired: true }, {
 	post() {
-		const findResult = findDirectMessageRoom(this.requestParams(), this.user);
+		const { username, usernames } = this.requestParams();
+
+		const users = username ? [username] : usernames && usernames.split(',').map((username) => username.trim());
+
+		if (!users) {
+			throw new Meteor.Error('error-room-not-found', 'The required "roomId", "username" or "usernames" param provided does not match any direct message');
```
Comment (rodrigok) [inline app/api/server/v1/im.js:40]: Where are you checking for the `roomId`?

Comment (ggazzo) [reply app/api/server/v1/im.js:40]: sorry why do we need a roomID? to create an im?
there is no reference about
https://rocket.chat/docs/developer-guides/rest-api/im/create/

Comment (rodrigok) [reply app/api/server/v1/im.js:40]: I don't know, you are referencing it in the error message 😄

Comment (ggazzo) [reply app/api/server/v1/im.js:40]: > Where are you checking for the roomId? there is no expected param called roomId
https://rocket.chat/docs/developer-guides/rest-api/im/create/

Comment (ggazzo) [reply app/api/server/v1/im.js]: I just kept the same message we were sending before. removed

**Pull request merged**